stdlib/LinearAlgebra/src/diagonal.jl

```diff
@@ -368,6 +368,15 @@ end
     return out
 end
 
+function (*)(Da::Diagonal, A::AbstractMatrix, Db::Diagonal)
+    _muldiag_size_check(Da, A)
+    _muldiag_size_check(A, Db)
+    dda = Da.diag
+    ddb = Db.diag
+    n = length(dda)
+    return broadcast(*, reshape(dda, n, 1), A, reshape(ddb, 1, n))
```

Comment:
```suggestion
    return broadcast(*, dda, A, reshape(ddb, 1, n))
```
save one allocation if `dda` is a `Vector`.

Reply:
does `reshape` allocate here?

Reply:
```julia
julia> a = [1,2,3];

julia> @btime reshape($a, 3, 1)
  24.774 ns (2 allocations: 96 bytes)
```